src/main/java/com/google/sps/servlets/ScheduledInterviewServlet.java

```diff
@@ -95,4 +110,62 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
   }
+
+  public List<ScheduledInterviewRequest> scheduledInterviewsToRequestObjects(
+      List<ScheduledInterview> scheduledInterviews, String timeZoneIdString) {
+    ZoneId timeZoneId;
+    if (timeZoneIdString == null) {
+      timeZoneId = ZoneId.systemDefault();
+    } else {
+      timeZoneId = ZoneId.of(timeZoneIdString);
+    }
+    List<ScheduledInterviewRequest> requestObjects = new ArrayList<ScheduledInterviewRequest>();
+    for (ScheduledInterview scheduledInterview : scheduledInterviews) {
+      requestObjects.add(makeSheduledInterviewRequest(scheduledInterview, timeZoneId));
+    }
+    return requestObjects;
+  }
+
+  private String getDateString(TimeRange when, ZoneId timeZoneId) {
+    LocalDateTime start = LocalDateTime.ofInstant(when.start(), timeZoneId);
+    LocalDateTime end = LocalDateTime.ofInstant(when.end(), timeZoneId);
+    String startTime = start.format(DateTimeFormatter.ofPattern("h:mm a"));
+    String endTime = end.format(DateTimeFormatter.ofPattern("h:mm a"));
+    String day = start.format(DateTimeFormatter.ofPattern("EEEE, MMMM d, yyyy"));
+    return String.format("%s from %s to %s", day, startTime, endTime);
+  }
+
+  private ScheduledInterviewRequest makeSheduledInterviewRequest(
+      ScheduledInterview scheduledInterview, ZoneId timeZoneId) {
+    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+    if (userId == null) {
```

Comment:
Q: when would userId ever be null?

Reply:
This is because there isn't a good enough mock for .getUserId() to not be null. Solely used for testing, Grant maybe copy over a comment each time this is called. This is what I've been using:
// Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
    // replaced with this hashcode.